lib/common/xml.c

```diff
@@ -1333,6 +1333,108 @@ replace_text(char *text, int index, size_t *length, const char *replace)
     return text;
 }
 
+/*!
+ * \internal
+ * \brief Replace special characters with their XML escape sequences
+ *
+ * XML allows the escaping of special characters by replacing them with entity
+ * references (for example, <tt>"&quot;"</tt>) or character references (for
+ * example, <tt>"&#13;"</tt>).
+ *
+ * The special characters <tt>'<'</tt> and <tt>'&'</tt> are not allowed in their
+ * literal forms in XML character data. Character data is non-markup text (for
+ * example, the content of a text node).
+ *
+ * Additionally, if an attribute value is delimited by single quotes, then
+ * single quotes must be escaped within the value. Similarly, if an attribute
+ * value is delimited by double quotes, then double quotes must be escaped
+ * within the value.
+ *
+ * For more details, see the "Character Data and Markup" section of the XML
+ * spec, currently section 2.4:
+ * https://www.w3.org/TR/xml/#dt-markup
+ *
+ * Pacemaker always delimits attribute values with double quotes, so this
+ * function doesn't escape single quotes.
+ *
+ * \param[in] text          Text to escape
+ * \param[in] escape_quote  If \c true, escape double quotes (should be enabled
+ *                          for attribute values)
+ *
+ * \return Newly allocated string equivalent to \p text but with special
+ *         characters replaced with XML escape sequences (or \c NULL if \p text
+ *         is \c NULL). If \p text is not \c NULL, the return value is
+ *         guaranteed not to be \c NULL.
+ *
+ * \note There are libxml functions that purport to do this:
+ *       \c xmlEncodeEntitiesReentrant() and \c xmlEncodeSpecialChars().
+ *       However, their escaping is incomplete. See:
+ *       https://discourse.gnome.org/t/intended-use-of-xmlencodeentitiesreentrant-vs-xmlencodespecialchars/19252
+ */
+char *
+pcmk__xml_escape(const char *text, bool escape_quote)
```

Comment:
worth a unit test

Reply:
I wish there already were one for `crm_xml_escape()` to use as a template... but you're right